lib/dogma.ex

```diff
@@ -8,20 +8,43 @@ defmodule Dogma do
 
   alias Dogma.Rules
   alias Dogma.ScriptSources
+  alias Dogma.Script
 
   @version Mix.Project.config[:version]
 
   def run(dir_or_file, config, dispatcher) do
     dir_or_file
-    |> ScriptSources.find(config.exclude)
-    |> ScriptSources.to_scripts
+    |> get_sources(config)
     |> notify_start(dispatcher)
     |> Rules.test(config.rules, dispatcher)
     |> notify_finish(dispatcher)
   end
 
+  def get_sources(dir_or_file, %{read_stdin: true} = _config) do
+    if dir_or_file == nil, do: dir_or_file="stdin"
+    case read_from_stdin do
+      {:ok, source} -> [source |> Script.parse(dir_or_file)]
+      _ -> System.halt(666)
+    end
+  end
+
+  def get_sources(dir_or_file, config) do
+    dir_or_file
+    |> ScriptSources.find(config.exclude)
+    |> ScriptSources.to_scripts
+  end
+
   def version, do: @version
 
+  defp read_from_stdin(source \\ "") do
+    case IO.read(:stdio, :line) do
```

Comment:
I was thinking to read the source in one go so its easier to add a check later to detect if the source is valid before we hand it off for parsing ?